### PR TITLE
Add guideline CustomTelemetry

### PR DESCRIPTION
**Pull request by ajkauffmann**
Linked to this discussion: https://github.com/microsoft/alguidelines/discussions/112
Text from Kennie is copied and slightly modified.

### CODE REVIEW

**Review by TheDoubleH — APPROVED** [review]

**Review by JeremyVyska — APPROVED** [review]
❤️

File: content/docs/BestPractices/CustomTelemetry/index.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+- documented
+- versioned
+- discoverable
+- non-breaking
+
+## Naming conventions and telemetry schema
+
+To make it easy for the consumer of telemetry to work with the data, please
+
+- use CamelCasing. This makes all fields in Application Insights look the same (signal logged through the AL LogMessage method will have "al" prefixed to dimension names.
```
Comment (waldo1001) [inline content/docs/BestPractices/CustomTelemetry/index.md:33]: Don't you mean PascalCasing?

camelCase <> PascalCase

File: content/docs/BestPractices/CustomTelemetry/index.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            DataClassification::SystemMetadata, 
+            TelemetryScope::ExtensionPublisher, 
+            CustomTelemetryDimensions);
+    end;
+end;
+```
+
+## Candidate data for telemetry
+
+Telemetry must be actionable for the customer. Do not emit signals that they cannot act on (knowing about CPU performance counters on the database is useless if the partner cannot scale the database)
```
Comment (waldo1001) [inline content/docs/BestPractices/CustomTelemetry/index.md:67]: May be put "actionable" in bolt .

File: content/docs/BestPractices/CustomTelemetry/index.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+Also, note that customers pay for data ingestion. So be mindful to not flood their telemetry resources. Consider to use TelemetryScope::ExtensionPublisher by default and only use TelemetryScope::All in case the customer can also act on the data.
+
+If you do not know where to start, consider using telemetry for deflection. In Dynamics 365 Business Central, they started with signal about authorization (successful/failed) to deflect support cases that was due to disabled users/wrong licenses.
+
+## How customers use telemetry
```
Comment (waldo1001) [inline content/docs/BestPractices/CustomTelemetry/index.md:73]: Is this chapter necessary - how is this "best practice"?

**Review by waldo1001 — CHANGES_REQUESTED** [review]

**Comment by waldo1001** [comment]
Would it be possible to include also a "bad code" example?

File: content/docs/BestPractices/CustomTelemetry/index.md (hunk 1 above)
Comment (ajkauffmann) [reply content/docs/BestPractices/CustomTelemetry/index.md:33]: Of course... The original word was 'PamelCasing' which seems to be a typo.

**Comment by ajkauffmann** [comment]
I've added an extra chapter about objects emitting telemetry signals.

**Review by waldo1001 — APPROVED** [review]

**Pull request merged**